extensions/amp-a4a/0.1/legacy-signature-verifier.js

```diff
@@ -66,13 +80,34 @@ export const AMP_SIGNATURE_HEADER = 'X-AmpAdSignature';
  * multiple Fast Fetch ad slots on a page (even ones from different ad networks)
  * to share the same cached public keys.
  *
+ * Experiment diversion between the new verifier and the legacy verifier occurs
+ * here.
+ *
  * @param {!Window} win
  * @return {!./signature-verifier.ISignatureVerifier}
  */
 export function signatureVerifierFor(win) {
   const propertyName = 'AMP_FAST_FETCH_SIGNATURE_VERIFIER_';
-  return win[propertyName] ||
-      (win[propertyName] = new LegacySignatureVerifier(win));
+  if (!win[propertyName]) {
+    if (/a4a-new-verifier/.test(win.location.hash)) {
+      forceExperimentBranch(win, VERIFIER_EXP_NAME, NEW_VERIFIER_EID);
+    } else if (/a4a-legacy-verifier/.test(win.location.hash)) {
+      forceExperimentBranch(win, VERIFIER_EXP_NAME, LEGACY_VERIFIER_EID);
+    } else {
+      randomlySelectUnsetExperiments(win, {
```

Comment:
You don't actually need to put this in the else given randomlySelectUnsetExperiments should do nothing if you've already forced the experiment on.